### PR TITLE
Added code to case insensitive search for demand title

### PR DESCRIPTION
**Pull request by ruksana2808**

### CODE REVIEW

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java [hunk 1]
```diff
@@ -406,6 +415,13 @@ public CustomResponse updateDemandStatus(JsonNode updateDetails, String token, S
                 ObjectMapper objectMapper = new ObjectMapper();
                 ObjectNode jsonNode = objectMapper.createObjectNode();
                 jsonNode.set(Constants.DEMAND_ID, new TextNode(saveJsonEntity.getDemandId()));
+                if (saveJsonEntity.getData().has(Constants.TITLE) && !saveJsonEntity.getData().get(Constants.TITLE).asText()
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: Please do a null check

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java [hunk 2]
```diff
@@ -197,11 +198,16 @@ public CustomResponse createDemand(JsonNode demandDetails, String token, String
             ObjectMapper objectMapper = new ObjectMapper();
             ObjectNode jsonNode = objectMapper.createObjectNode();
             jsonNode.set(Constants.DEMAND_ID, new TextNode(saveJsonEntity.getDemandId()));
+            if (saveJsonEntity.getData().has(Constants.TITLE) && !saveJsonEntity.getData().get(Constants.TITLE).asText()
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: Add a null check

**Review by Sahil-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java (hunk 1 above)
Comment (ruksana2808) [reply src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: made changes

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java (hunk 2 above)
Comment (ruksana2808) [reply src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: made changes

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java [hunk 3]
```diff
@@ -198,12 +198,15 @@ public CustomResponse createDemand(JsonNode demandDetails, String token, String
             ObjectMapper objectMapper = new ObjectMapper();
             ObjectNode jsonNode = objectMapper.createObjectNode();
             jsonNode.set(Constants.DEMAND_ID, new TextNode(saveJsonEntity.getDemandId()));
-            if (saveJsonEntity.getData().has(Constants.TITLE) && !saveJsonEntity.getData().get(Constants.TITLE).asText()
-                .isEmpty()) {
+            if (saveJsonEntity.getData().has(Constants.TITLE) && !saveJsonEntity.getData()
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: saveJsonEntity add for everyObject null pointer which we think probably with null and do proper error handling

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java [hunk 4]
```diff
@@ -415,12 +418,15 @@ public CustomResponse updateDemandStatus(JsonNode updateDetails, String token, S
                 ObjectMapper objectMapper = new ObjectMapper();
                 ObjectNode jsonNode = objectMapper.createObjectNode();
                 jsonNode.set(Constants.DEMAND_ID, new TextNode(saveJsonEntity.getDemandId()));
-                if (saveJsonEntity.getData().has(Constants.TITLE) && !saveJsonEntity.getData().get(Constants.TITLE).asText()
-                    .isEmpty()) {
+                if (saveJsonEntity.getData().has(Constants.TITLE) && !saveJsonEntity.getData()
```
Comment (Sahil-tarento) [inline src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: saveJsonEntity add for everyObject null pointer which we think probably with null and do proper error handling

**Review by Sahil-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java (hunk 3 above)
Comment (ruksana2808) [reply src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: @Sahil-tarento made null pointer check for saveJsonEntity . can you please review

File: src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java (hunk 4 above)
Comment (ruksana2808) [reply src/main/java/com/igot/cb/demand/service/DemandServiceImpl.java]: Added null pointer check for saveJsonEntity

**Pull request merged**